patterns/Facade-Pattern/README.md

```diff
@@ -0,0 +1,113 @@
+
+## Uses of Facade Design Pattern
+1. Facade offers a higher-level interface, which makes interacting with complex subsystems easier.
+2. Facade serves as an API between the presentation layer and service layer in multi-layer web applications, concealing the complex nature of business objects and interactions.
+3. Facade can be used to modify complicated or confusing legacy code by hiding its complexities behind a simpler interface, making it simpler for clients to work with.
+4. Facade is frequently used with other design patterns, such as Singleton or Factory, to improve usability and provide a consistent interface for client interaction.
+(Poorte, 2023)
+
+## Diagram
+![javascript-facade](https://github.com/nazrinzuwair/pattern-library/assets/144160072/757bff23-d434-4189-8006-db5b1415989e)
+
+The elements that are involved in the Facade design pattern are:
+1. Subsystem: The entire intricate logic and implementation are handled by a class, collection of classes, or interfaces that make up the Subsystem.
+2. Facade: The class or object that acts as a doorway into the Subsystem's implementation while concealing its complex details of its implementation.
+3. Client: The Client uses the streamlined Facade Interface for performing operations without being aware of the specifics of the internal implementation.
+(GeeksforGeeks, 2023)
+
+## Example demonstrating the Facade Method pattern in JavaScript:
+
+### Step 1: Creation of Subsystem
+
+In JavaScript, we first define classes representing different subsystems. Each class encapsulates functionality related to a specific aspect of the system:
+
+``` javascript
+// Define a class for SoftwareDeveloper
+class SoftwareDeveloper {
+    // Define a method within the SoftwareDeveloper class to represent coding
+    code() {
+      console.log('Software Developer codes the frontend and backend for an application');
+    }
+}
+
+// Define a class for Designer
+class Designer {
+    // Define a method within the Designer class to represent designing
+    design() {
+      console.log('UI Designer creates the wireframes for the website.');
+    }
+}
+
+// Define a class for DatabaseEngineer
+class DatabaseEngineer {
+    // Define a method within the DatabaseEngineer class to represent connecting
+    connect() {
+      console.log('Database engineer performs the connection between the application and the database.');
+    }
+}
+```
+
+We have three classes here: SoftwareDeveloper, Designer, and DatabaseEngineer, each of which represents a subsystem responsible for a specific task in the software development process. The SoftwareDeveloper class's code() method takes handle of coding activities, the Designer class's design() method takes responsibility of designing tasks, and the DatabaseEngineer class's connect() method takes charge of database connection tasks. Each subsystem's behavior is specified by these classes.
+
```

Comment:
Any time you reference code in a sentence or paragraph you can use in-line code markdown (i.e. single backticks) to denote that you are referring to code. For example, the paragraph above should be styled as follows:

We have three classes here: `SoftwareDeveloper`, `Designer`, and `DatabaseEngineer`, each of which represents a subsystem responsible for a specific task in the software development process. The `SoftwareDeveloper` class's `code()`method takes handle of coding activities, the `Designer` class's `design()` method takes responsibility of designing tasks, and the `DatabaseEngineer` class's `connect()` method takes charge of database connection tasks. Each subsystem's behavior is specified by these classes.

Reply:
Oh! I see you've done that below - you must have just missed this paragraph.